base/src/main/scala/data/DisjunctionFunctions.scala

```diff
@@ -1,10 +1,17 @@
 package scalaz
 package data
 
+import Disjunction.{\/, \/-, -\/}
+
 trait DisjunctionFunctions {
-  @inline def -\/[L](value: L): Disjunction[L, Nothing] = Disjunction.-\/(value)
-  @inline def \/-[R](value: R): Disjunction[Nothing, R] = Disjunction.\/-(value)
+  @inline def -\/[L](value: L): Disjunction[L, Nothing] = -\/(value)
```

Comment:
All right, so I'll merge this PR as is, and we'll go with a no-variance approach.

I feel like by having proper syntax we can minimize the cost of being explicit about both side so we can at least be explicit only once.

(I have in mind for example, introducing some `BoolOps` to make the case of `if (bool) left(a) else right(b)` digestible)